### DISCUSSION TITLE
Get right indicator search term.

### DISCUSSION THREAD

**Discussion by Rohianon** [owid/etl#2354] [Q&A] [answered]
Hi team, 

i am seeking to find the right search term that i can be able to query for these two indicators. 

The data needs to match as the one on the grapher. 

1. Nitrous Oxide Emissions - https://ourworldindata.org/grapher/nitrous-oxide-emissions?tab=chart
2. CO2 Emissions - https://ourworldindata.org/grapher/annual-co2-emissions-per-country

kindly help.

**Comment by lucasrodes** [comment, marked as answer]
Hi @Rohianon,
Thanks for your interest in our project.

I am assuming that you are looking for the data itself?

If so, use our [python API](https://docs.owid.io/projects/etl/api/). For that, you need the identifiers for both indicators, which are:

1. Nitrous Oxide Emissions: `emissions/2023-11-23/national_contributions/national_contributions#annual_emissions_n2o_total_co2eq`
2. CO2 Emissions: `gcp/2023-12-12/global_carbon_budget/global_carbon_budget#emissions_total`

Then, you can use:

```python
from owid.catalog import RemoteCatalog

catalog = RemoteCatalog()

tb_1 = catalog.find(
    namespace="emissions",
    version="2023-11-23",
    dataset="national_contributions",
    table="national_contributions",
).load()
tb_n2o = tb_1[["annual_emissions_n2o_total_co2eq"]]

tb_2 = catalog.find(
    namespace="gcp",
    version="2023-12-12",
    dataset="global_carbon_budget",
    table="global_carbon_budget",
).load()
tb_co2 = tb_2[["emissions_total"]]
```

This should load the indicators that you mention as two tables: `tb_n2o` and `tb_co2`. Note that these are `catalog.Table` objects, which are identical to DataFrames, **but also contain metadata** (at table and indicator level).

**Access table metadata:** 
```python
tb_co2.metadata
```

**Access indicator metadata:** 
```python
tb_co2["emissions_total"].metadata
```

Note that the metadata renders nicely in Jupyter notebooks and can be converted into a dictionary: `df_co2["emissions_total"].metadata.to_dict()`. 

More on metadata [here](https://docs.owid.io/projects/etl/api/python/#metadata).

**Comment by Rohianon** [comment]
This was super accurate for sure. 
![image](https://github.com/owid/etl/assets/41194018/2b2e9e9f-2396-4d1e-87bc-f50da651877d)
![image](https://github.com/owid/etl/assets/41194018/3cd82dc3-e9fe-4b0d-8824-8e277cdb3738)


I expected that the indicators would become a hard deal to find as I struggled finding them. 

Maybe a suggestion would be to have a documentation for the mapping of indicators to the grapher? 
If it already exists, I'd be glad to be pointed out to it.
I had to do a lot of guess work with the `catalog.find()` method to be able to know what indicators I was looking for and that they were actually the correct ones.

**Reply by lucasrodes** [reply]
Agree, @Rohianon. I think that would be a very nice addition to our Grapher charts.

We have been in discussions on how to surface that there. Will keep you posted!

**Reply by Rohianon** [reply]
@lucasrodes, 

Thanks you so much. I really appreciate your aid on this.

I only have one more indicator left: 
-  [Per capita electricity generation from solar](https://ourworldindata.org/grapher/solar-electricity-per-capita) 
Tried it out but was only able to find `solar_electricity_generation_twh` instead. 

Will be glad to hear your aid again on the same.

**Reply by lucasrodes** [reply]
`energy/2023-12-12/electricity_mix/electricity_mix#per_capita_solar_generation__kwh`

meaning:
- namespace: `energy`
- version: `2023-12-12`
- dataset: `electricity_mix`
- table: `electricity_mix`

and column `per_capita_solar_generation__kwh`

**Reply by Rohianon** [reply]
Thanks you, this is so helpful and I really appreciate. 

Will definitely add you more names incase I am stuck on finding any indicators.

Cheers!

**Comment by lucasrodes** [comment]
@Rohianon FYI, we are working to expose a new API so that you can get the relevant indicators for a given chart. More details here: https://github.com/owid/etl/pull/2629
